File: JerishwinJoseph-Assessment-8/Assessment/src/test/java/com/actions/LoginAction.java
package com.actions;

import java.time.Duration;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import com.pages.LoginPage;


public class LoginAction extends BaseAction{
	
	WebDriverWait wait;
	LoginPage lp;
	
	public LoginAction(WebDriver driver) {
		super(driver);
		wait = new WebDriverWait(driver, Duration.ofSeconds(10));
		lp = new LoginPage(driver);
	}

	public void login(String email, String password) {
        wait.until(ExpectedConditions.elementToBeClickable(lp.myAccountDropdown)).click();

        wait.until(ExpectedConditions.elementToBeClickable(lp.loginLink)).click();

        wait.until(ExpectedConditions.visibilityOf(lp.emailInput));
        
        lp.emailInput.sendKeys(email);

        lp.passwordInput.sendKeys(password);

        lp.loginButton.click();

        wait.until(ExpectedConditions.urlContains("account"));
    }

}

File: JerishwinJoseph-Assessment-11/Task2/action/login_action.py
from action.base_action import BaseAction
from page.login_page import LoginPage


class LoginAction(BaseAction):

    def __init__(self, driver):
        super().__init__(driver)

    def login(self, email, password):

        self.enter_text(LoginPage.EMAIL, email)
        self.enter_text(LoginPage.PASSWORD, password)

        self.click(LoginPage.LOGIN_BUTTON)

    def get_warning_message(self):

        return self.get_text(
            LoginPage.WARNING_MSG
        )